### ISSUE TITLE
LE Audio: Media: Pass structs by reference and not value

### ISSUE THREAD

**Issue by Thalley** [zephyrproject-rtos/zephyr#42096]
**Is your enhancement proposal related to a problem? Please describe.**
There are a few structs in the media implementation where the structs are passed by value, instead of reference. In smaller structs that may be OK, but there are structs > sizeof(pointer), where it definitely shouldn't be the case. Furthermore, for consistency we may want to just pass all structs, regardless of size, by reference instead of value. 

Specific examples are the `struct mpl_cmd_ntf` and `struct mpl_cmd`, but there may be others. 

**Describe the solution you'd like**
Basically change uses of the structs from e.g. 
```c
typedef void (*bt_mcc_cmd_ntf_cb)(struct bt_conn *conn, int err, struct mpl_cmd_ntf ntf);
```
to
```c
typedef void (*bt_mcc_cmd_ntf_cb)(struct bt_conn *conn, int err, struct mpl_cmd_ntf *ntf);
```

**Describe alternatives you've considered**
In some cases, we may just want to skip the smaller struct and just pass the fields, but that's doubtful. 

**Additional context**
N/A
